Version 1.1

barunio opened the pull request:
Version 1.1 mostly adds functionality to allow controllers and views to access public context methods directly, via `method_missing` and `view_helpers`. It also adds a descriptive README and cleans up the gem a bit.

fixlr commented on README.md:
```diff
@@ -1 +1,193 @@
+  def current_user
+    User.find_by(id: session[:user_id])
+  end
+  memoize :user
```
```suggestion
  memoize :current_user
```

fixlr reviewed (approved):
Not blockers for this, but I'd love to see a simple `CHANGELOG.md` added, and to have tags pushed for each of the versions.

barunio committed:
add CHANGELOG file

barunio committed:
add method_missing logic to Context::Controller

This commit adds functionality to easily access context methods in
controllers. This is done through the use of method_missing.

Prior to this commit, if a controller wanted to make use of methods
defined in contexts, it would need to do so explicitly, like so:
  @__context.mymethod
Now, if `mymethod` is a public method available in the context stack,
the controller can simply call it directly without needing to
reference `@__context`.

barunio committed:
rename @context to @__context in controller

It is possible that an application using this gem will want to use an
instance variable named `@context` in a controller for unrelated
purposes. If that were to happen, it would break this gem’s code.
This commit changes the variable name used in the
`Context::Controller` module to `@__context` to make that sort of
issue very unlikely.

barunio committed:
bump version to 1.1.0

barunio committed:
use require_relative instead of require

barunio committed:
fix gem dependencies

This commit fixes some gem dependencies that were copy/pasted from a
different gem when creating this repo initially.

The Rails dependency is set to > 4.0.2 because that’s when
`before_filter` was changed to `before_action`, and it’s unlikely that
anyone will need to use this gem for applications using versions of
Rails earlier than that.

There aren’t any tests that rely on ActiveRecord, so the AR
environement variable is not relevant.

barunio committed:
add BaseContextHelper module

This module adds the necessary functionality for views to access
view_helper methods defined in the context stack. It does this via
method_missing logic. The intended usage of this module is for
`ApplicationController` to explicitly declare
`helper Context::BaseContextHelper`.

barunio committed:
add README content

barunio commented:
@fixlr Good call on the changelog -- I meant to do that but had forgotten. Added now.
